File: 3276-MinimumNumberOfPushesToTypeWordIi/3276-MinimumNumberOfPushesToTypeWordIi.java
// Last updated: 7/1/2025, 11:03:15 PM
import java.util.*;

class Solution {
    public int minimumPushes(String word) {
        Integer counter[] = new Integer[26]; 
        int res = 0; 

        Arrays.fill(counter, 0);

        for (char character : word.toCharArray()) {
            counter[(int) character - (int) 'a'] += 1;
        }

        Arrays.sort(counter, Collections.reverseOrder());

        for (int i = 0; i < 26; i++) {
            int count = counter[i];
            res += count * ((i / 8) + 1); 
            System.out.println(res);
        }

        return res; 
    }
}

File: 3276-MinimumNumberOfPushesToTypeWordIi/3276-MinimumNumberOfPushesToTypeWordIi.py
# Last updated: 7/1/2025, 11:03:19 PM
class Solution:
    def minimumPushes(self, word: str) -> int:
        counter = [0] * 26
        minimum_press = 0
        for i in word: 
            counter[ord(i) - ord("a")] += 1
        
        counter.sort(reverse=True)
        for i in range(26): 
            minimum_press += counter[i] * (int(i/8) + 1) 
        return minimum_press

        